src/builder/build_traits.rs

```diff
@@ -675,9 +675,9 @@ fn wire_up<T: Dataflow + ?Sized>(
             if !typ.copyable() {
                 let val_err: ValidationError = InterGraphEdgeError::NonCopyableData {
                     from: src,
-                    from_offset: src_port,
+                    from_offset: src_port.into(),
```

Comment:
Possibly we could extend `NonCopyableData` to take an OutgoingPort and an IncomingPort, etc., etc. - by all means shout out cases like that, I am not claiming to have found them all